portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,35 +15,69 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
 import java.util.ArrayList;
+import java.util.Date;
 import java.util.List;
-import com.google.gson.Gson;
+
+import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
-  private List<String> comments;
-
-  @Override
-  public void init() {
-    comments = new ArrayList<>();
-    comments.add("London");
-    comments.add("Tokyo");
-    comments.add("Abuja");
-  }
+  private DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+  private Key homepageCommentKey = KeyFactory.createKey("Comments", "homepage comments");
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Query query = new Query("Comment").setAncestor(homepageCommentKey)
+                      .addSort("timestamp", SortDirection.DESCENDING);
+    PreparedQuery results = datastore.prepare(query);
+    List<Comment> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String author = (String) entity.getProperty("author");
```

Comment:
i would suggest having two helper functions to do conversions:
Comment -> Entity
Entity -> Comment

Reply:
Fixed in [this commit](https://github.com/ifeomi/step/pull/13/commits/fb32d16655adbce3331bfd72e3dbc8cfd476be83) on PR #13 since that's the earliest occurrence :)